### PR TITLE
Rewrote the project with more modern C++ features

### PR DESCRIPTION
**Pull request by TerensTare**
Used if-initializer, direct curly braces initalization, braced-initialization for all clases members, added noexcept where needed. Also updated copyright year.

### CODE REVIEW

File: TMXLoader/TMXMap.cpp [hunk 1]
```diff
@@ -14,165 +14,96 @@
...
-{
-    for (unsigned int index = 0; index < m_layerVector.size(); ++index)
-    {
-        if (m_layerVector[index].getName() == layerName)
-        {
-            return &m_layerVector[index];
-        }
-    }
-    
+    for (unsigned idx{0}; idx < m_layerVector.size(); ++idx)
```
Comment (bjorn) [inline TMXLoader/TMXMap.cpp]: Since we're using modern C++, I'd suggest to use range-based for as well:
```c++
for (const auto &layer : m_layerVector)
```

Comment (TerensTare) [reply TMXLoader/TMXMap.cpp]: That is another good change. It's just that I'm not used to range-based for yet.

**Comment by bjorn** [comment]
Not sure if it was intentional, but 508b0cbec6072c24d28a26a893346bdedf158ac2 reverted many of the earlier changes.

**Comment by martygrant** [comment]
Oops! I'll fix this later, thanks for pointing that out @bjorn

**Comment by TerensTare** [comment]
I'll fix that in a minute @martygrant .

**Pull request merged**